parsons/quickbooks/quickbooks.py

```diff
@@ -0,0 +1,734 @@
+                    'yes', 'no', or 'both'. Default is 'yes'.
+
+                manager_ids: Int
+                    Comma separated list of one or more manager ids you'd like to filter on.
+
+            name: String
+                Comma separated list of one or more group names you'd like to filter on.
+
+            modified_before: String
+                Only groups modified before this date/time will be returned,
+                in ISO 8601 format (YYYY-MM-DDThh:mm:ss±hh:mm)
+
+            modified_since: String
+                Only groups modified since this date/time will be returned,
+                in ISO 8601 format (YYYY-MM-DDThh:mm:ss±hh:mm)
+
+            supplemental_data: String
+                'yes' or 'no'. Default is 'yes'.
+                Indicates whether supplemental data should be returned.
+
+            per_page: Int
+                Represents how many results you'd like to retrieve per request (page).
+                Default is 50. Max is 50.
+
+            page: Int
+                Represents the page of results you'd like to retrieve.
+
+        `Returns:`
+            Parsons Table
+        """
+
+        querystring = {
+            "ids": ids,
+            "active": active,
+            "manager_ids": manager_ids,
+            "name": name,
+            "modified_before": modified_before,
+            "modified_since": modified_since,
+            "supplemental_data": supplemental_data,
+            "per_page": per_page,
+        }
+
+        logger.info("Retrieving groups.")
+        tbl = self.qb_get_request(end_point="groups", querystring=querystring)
+
+        logger.info(f"Found {tbl.num_rows} groups.")
+        if tbl.num_rows > 0:
+            return tbl
+        else:
+            return Table()
+
+    def get_jobcodes(
+        self,
+        ids=None,
+        parent_ids=None,
+        name=None,
+        type=None,
+        active=None,
+        customfields=None,
+        modified_before=None,
+        modified_since=None,
+        supplemental_data=None,
+        per_page=None,
+        page=None,
+    ):
+        """
+        This function allows you to call the /jobcodes endpoint of the Quickbooks Time API.
+
+        `Args:`
+            ids: Int
+                Comma separated list of one or more jobcode ids you'd like to filter on.
+                Only jobcodes with an id set to one of these values will be returned.
+                If omitted, all jobcodes matching other specified filters are returned.
+
+            parent_ids: Int
+                Default is -1 (meaning all jobcodes will be returned regardless of parent_id).
+                Comma separated list of one or more jobcode parent_ids you'd like to filter on.
+                Only jobcodes with a parent_id set to one of these values will be returned.
+                Additionally you can use 0 to get only the top-level jobcodes.
+
+                Then get the id of any results with has_children=yes
```

Comment:
The formatting of this looks a little weird. Have you built the docs and checked that it looks okay?